contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.cc

```diff
@@ -170,31 +169,24 @@ ssl_private_key_result_t rsaPrivateKeySignInternal(CryptoMbPrivateKeyConnection*
     return status;
   }
 
+  // Add RSA padding to the the hash. Only `PSS` padding is supported.
+  // TODO(ipuustin): add PKCS #1 v1.5 padding scheme later if requested.
+  if (!SSL_is_signature_algorithm_rsa_pss(signature_algorithm)) {
+    ops->logDebugMsg("non-supported RSA padding");
+    return status;
```

Comment:
Thanks, that makes sense and I agree with you that it would be nicer that way. But on the other hand I would like to keep this as close as possible to the patch that I'm backporting. Would it be ok if we rather improved the code in the main branch with a future PR?